hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/state/singleton/EntityIdSingleton.java

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package com.hedera.mirror.web3.state.singleton;
+
+import static com.hedera.node.app.ids.schemas.V0490EntityIdSchema.ENTITY_ID_STATE_KEY;
+
+import com.hedera.hapi.node.state.common.EntityNumber;
+import com.hedera.mirror.common.domain.entity.EntityId;
+import com.hedera.mirror.web3.evm.properties.MirrorNodeEvmProperties;
+import com.hedera.mirror.web3.repository.EntityRepository;
+import com.hedera.node.config.data.HederaConfig;
+import jakarta.inject.Named;
+import lombok.RequiredArgsConstructor;
+
+@Named
+@RequiredArgsConstructor
+@SuppressWarnings("deprecation")
+public class EntityIdSingleton implements SingletonState<EntityNumber> {
+    private final EntityRepository entityRepository;
+    private final MirrorNodeEvmProperties mirrorNodeEvmProperties;
+
+    @Override
+    public String getKey() {
+        return ENTITY_ID_STATE_KEY;
+    }
+
+    @Override
+    public EntityNumber get() {
+        final long firstUserEntity = mirrorNodeEvmProperties
+                .getVersionedConfiguration()
+                .getConfigData(HederaConfig.class)
+                .firstUserEntity();
+        final var maxId = EntityId.of(entityRepository.findMaxId());
```

Comment:
I think for non-zero realm project we should filter by shard and realm as well.
```suggestion
        final var maxId = EntityId.of(entityRepository.findMaxId(commonProperties.getShard(), commonProperties.getRealm()));
```

Reply:
Using common properties now, modified the query to work with shard and realm